integration_tests/src/main/python/avro_test.py

```diff
@@ -30,46 +30,56 @@
     'spark.rapids.sql.format.avro.enabled': 'true',
     'spark.rapids.sql.format.avro.read.enabled': 'true'}
 
+rapids_reader_types = ['PERFILE', 'COALESCING']
 
-@pytest.mark.parametrize('gen', support_gens)
-@pytest.mark.parametrize('v1_enabled_list', ["avro", ""])
-def test_basic_read(spark_tmp_path, gen, v1_enabled_list):
+@pytest.mark.parametrize('v1_enabled_list', ["avro", ""], ids=["v1", "v2"])
+@pytest.mark.parametrize('reader_type', rapids_reader_types)
+def test_basic_read(spark_tmp_path, v1_enabled_list, reader_type):
+    gen_list = [('_c' + str(i), gen) for i, gen in enumerate(support_gens)]
     data_path = spark_tmp_path + '/AVRO_DATA'
+    # 50 files for the coalescing reading case
     with_cpu_session(
-        lambda spark: unary_op_df(spark, gen).write.format("avro").save(data_path)
+        lambda spark: gen_df(spark, gen_list).repartition(50).write.format("avro").save(data_path)
     )
 
     all_confs = copy_and_update(_enable_all_types_conf, {
+        'spark.rapids.sql.format.avro.reader.type': reader_type,
         'spark.sql.sources.useV1SourceList': v1_enabled_list})
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark: spark.read.format("avro").load(data_path),
         conf=all_confs)
 
 
-@pytest.mark.parametrize('v1_enabled_list', ["", "avro"])
-def test_avro_simple_partitioned_read(spark_tmp_path, v1_enabled_list):
+@pytest.mark.parametrize('v1_enabled_list', ["", "avro"], ids=["v1", "v2"])
+@pytest.mark.parametrize('reader_type', rapids_reader_types)
+def test_avro_simple_partitioned_read(spark_tmp_path, v1_enabled_list, reader_type):
     gen_list = [('_c' + str(i), gen) for i, gen in enumerate(support_gens)]
     first_data_path = spark_tmp_path + '/AVRO_DATA/key=0/key2=20'
     with_cpu_session(
-        lambda spark: gen_df(spark, gen_list).write.format("avro").save(first_data_path))
+        lambda spark: gen_df(spark,
+            gen_list).repartition(50).write.format("avro").save(first_data_path))
```

Comment:
So what's the value you would suggest ?  
Personally `repartition` is easier than setting maxPartitionBytes since we need to know the total generated data size (which is not easy to get) to make sure it will produce enough partitions. Also if we add more types here, the actual partition number will change. While `repartition` can give the exact partition number we want.